### PR TITLE
Added comments to the Rakefile

### PR DESCRIPTION
**Pull request by sarangan12**
This is to address part of the task of https://github.com/Azure/azure-sdk-for-ruby/issues/1075. The rest of the task will be handled in wiki pages. Please review.

@veronicagg @salameer Please review and approve.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -12,6 +12,102 @@
...
+# This rakefile consists of several tasks. This section will provide an overview of the
+# tasks and their dependencies.
+#
+# -------------------------------------------------------------------------------------------------------------------------------------------
+# | No |             Task                | Description                                                                                      |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 1  | regen                           | Regenerates all the SDK versions for all the services and all the profiles for all the services. |
+# |    |                                 | This task has regen_sdk_versions and regen_all_profiles as prereqs.                              |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 2  | regen_sdk_versions              | Regenerates all the SDK versions for all the services. At the end, this task calls               |
```
Comment (veronicagg) [inline Rakefile]: I would probably mention that this task it focused on autorest code generation and will generate api versions as indicated by config.json file

File: Rakefile [hunk 2]
```diff
@@ -12,6 +12,102 @@
...
+# |    |                                 | regen_individual_require_files as an enhancement. This task has clean_generated as prereq.       |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 3  | regen_all_profiles              | Regenerates all the profiles for all the services. This task has regen_rollup_profile and        |
```
Comment (veronicagg) [inline Rakefile]: I would indicate that this task uses a specific generator - separate from autorest, and that it has dependencies on certain json files like profiles.json

File: Rakefile [hunk 3]
```diff
@@ -12,6 +12,102 @@
...
+# |    |                                 | regen_individual_profiles as prereqs.                                                            |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 4  | clean_generated                 | Cleans all the generated folders in multiple SDK versions.                                       |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 5  | regen_rollup_profile            | Regenerates all profiles within the rollup gem. At the end, this task calls                      |
```
Comment (veronicagg) [inline Rakefile]: I would call out the requirement of having a proper definition of the profile in the profile.json file

File: Rakefile [hunk 4]
```diff
@@ -12,6 +12,102 @@
...
+# |    |                                 | regen_rollup_require_files as an enhancement. This task has clean_rollup_profiles as prereq.     |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 6  | regen_individual_profiles       | Regenerates all profiles with all the services. At the end, this task calls                      |
```
Comment (veronicagg) [inline Rakefile]: same comment as above

File: Rakefile [hunk 5]
```diff
@@ -12,6 +12,102 @@
...
+# | 7  | clean_all_profiles              | Cleans all profiles inside the rollup gem and individual gems.This task has clean_rollup_profiles|
+# |    |                                 | and clean_individual_profiles as prereqs.                                                        |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 8  | clean_rollup_profiles           | Cleans all profiles inside the rollup gem.                                                       |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 9  | clean_individual_profiles       | Cleans all profiles inside individual gems.                                                      |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 10 | regen_all_require_files         | Regenerates the require files for the rollup gem and individual gems. This task has              |
```
Comment (veronicagg) [inline Rakefile]: I don't think it's clear what "require_files" are, can we add a definition of that somewhere? or a pointer to an example? you could add a reference here like (*) or (1) and explain below

File: Rakefile [hunk 6]
```diff
@@ -12,6 +12,102 @@
...
+# |    |                                 | regen_rollup_require_files and regen_individual_require_files as prereqs.                        |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 11 | regen_rollup_require_files      | Regenerates the require files for the rollup gem                                                 |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 12 | regen_individual_require_files  | Regenerates the require files for the individual gems                                            |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 13 | release                         | Release gems. This task has build as a prereq.                                                   |
```
Comment (veronicagg) [inline Rakefile]: we should indicate what controls what gets released when running this task - input file used. And indicate... how would we release an individual gem? btw, based on another issue I was recently looking at, what will be the story for "tagging" the repo when we ship individual gems?

File: Rakefile [hunk 7]
```diff
@@ -12,6 +12,102 @@
...
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 14 | build                           | Builds all gems. This task has clean as a prereq.                                                |
```
Comment (veronicagg) [inline Rakefile]: similarly here, what controls what gets build should be indicated.

File: Rakefile [hunk 8]
```diff
@@ -12,6 +12,102 @@
...
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 15 | clean                           | Cleans the pks folder inside  all the gems.                                                      |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 16 | bundle:update                   | Performs bundle update for all gems.                                                             |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 17 | spec                            | Executes tests for all gems.                                                                     |
+# -------------------------------------------------------------------------------------------------------------------------------------------
+#
+#                                                       regen
+#                                                         |
```
Comment (veronicagg) [inline Rakefile]: nice diagram!

File: Rakefile (hunk 1 above)
Comment (sarangan12) [reply Rakefile]: Done

File: Rakefile (hunk 2 above)
Comment (sarangan12) [reply Rakefile]: Done

File: Rakefile (hunk 3 above)
Comment (sarangan12) [reply Rakefile]: Done

File: Rakefile (hunk 4 above)
Comment (sarangan12) [reply Rakefile]: Done

File: Rakefile (hunk 5 above)
Comment (sarangan12) [reply Rakefile]: Done for the first one. WIP on CI modification

File: Rakefile (hunk 6 above)
Comment (sarangan12) [reply Rakefile]: Done for the first one. WIP on CI modification

File: Rakefile (hunk 7 above)
Comment (sarangan12) [reply Rakefile]: build task builds for all gems.

File: Rakefile (hunk 8 above)
Comment (sarangan12) [reply Rakefile]: Tks

**Comment by sarangan12** [comment]
All comments addressed. Merging

**Pull request merged**